implementation/src/main/java/io/smallrye/mutiny/groups/UniMemoize.java

```diff
@@ -50,9 +50,27 @@ public Uni<T> until(BooleanSupplier invalidationGuard) {
      *        {@code null}, must be strictly positive
      * @return a new {@link Uni}
      * @apiNote This is an experimental API
+     * @deprecated use {@link forDuration(Duration)} instead
      */
     @CheckReturnValue
     public Uni<T> atLeast(Duration duration) {
+        return forDuration(duration);
+    }
+
+    /**
+     * Memoize the received item or failure for a duration after the upstream subscription has been received.
+     * <p>
+     * New subscribers will receive the memoized item or failure.
+     * When duration has elapsed then the first subscription causes a new upstream subscription, and the next
+     * subscribers get a chance to observe new values.
+     *
+     * @param duration the memoization duration after having received the subscription from upstream, must not be
+     *        {@code null}, must be strictly positive
+     * @return a new {@link Uni}
+     * @apiNote This is an experimental API
+     */
+    @CheckReturnValue
+    public Uni<T> forDuration(Duration duration) {
```

Comment:
I'm not sure the semantics are clear from the name.

- `atLeast` means that when the duration has elapsed, then the memoization is still in effect,
- `forDuration` looks more like `atMost` IMHO

Reply:
Yeah, seems `atLeast` right name for this method